tests/geolocation/test_geolocation_shared_via_w3c_api.py

```diff
@@ -0,0 +1,161 @@
+import pytest
+from selenium.common import NoSuchElementException
+from selenium.webdriver import Firefox
+from selenium.webdriver.common.by import By
+
+from modules.browser_object import Navigation, TabBar
+from modules.page_object import GenericPage
+
+
+@pytest.fixture()
+def test_case():
+    return "15186"
+
+
+@pytest.fixture()
+def set_prefs():
+    """Set prefs"""
+    return [
+        (
+            "geo.provider.network.url",
+            "https://www.googleapis.com/geolocation/v1/geolocate?key"
+            "=%GOOGLE_LOCATION_SERVICE_API_KEY%",
+        )
+    ]
+
+
+TEST_URL = "https://www.w3schools.com/html/html5_geolocation.asp"
+
+
+def handle_cookie_banner(driver, web_page):
+    """
+    Address the cookie banner manually if appears, as the cookie banner dismissal preference is not effective in this
+    context
+    """
+    try:
+        driver.switch_to.window(driver.window_handles[-1])
+        web_page.find_element(By.ID, "accept-choices").click()
+    except NoSuchElementException:
+        # If the cookie banner is not found, continue with the test
+        pass
+
+
+def handle_geolocation_prompt(nav, button_type="primary"):
+    """
+    Handles geolocation prompt by clicking either the 'Allow' or 'Block' button based on the button_type provided
+    """
+    button_selector = f"popup-notification-{button_type}-button"
+    nav.element_clickable(button_selector)
+    nav.click_on(button_selector)
+
+
+def click_geolocation_button_trigger(web_page):
+    """
+    Clicks the 'Try It' button inside the webpage to trigger the geolocation prompt
+    """
+    geolocation_button_selector = (
+        "button.w3-btn.w3-blue.w3-round[onclick='getLocation()']"
+    )
+    web_page.find_element(By.CSS_SELECTOR, geolocation_button_selector).click()
+
+
+def is_location_marker_displayed(web_page):
+    """
+    Checks if the location marker ('You are here!') is displayed on the web page inside the map.
+    """
+    location_marker = web_page.find_elements(
+        By.CSS_SELECTOR, "div[aria-label='You are here!'][role='img']"
+    )
+
+    if location_marker:  # Check if the marker exists
+        return location_marker[0].is_displayed()
+    else:
+        return False
+
+
+def open_web_page_in_new_tab(web_page, driver, tabs, expected_num_tabs):
```

Comment:
could you move this to page_base.py? I think this would be useful for multiple tests...

Reply:
I've moved it for now  to browser_object_tabbar instead. I might be mistaken, but I believe it's more appropriate there.